### PR TITLE
[pipelining] Follow improvements in export.unflatten

### PR DESCRIPTION
**Pull request by kwen2501**
Previously, we make a copy of `torch.export.unflatten` in pippy/_unflatten.py.

But it turns out to be too hard to track bug fixes and improvements in upstream version. For example, `torch.export.unflatten` recently added support for tied parameters, which is something pipelining needs.

Now that we moved into pytorch, we make a reference to `torch.export.unflatten` instead of maintaining a copy.

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #126217

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @chauhang @d4l3k

### CODE REVIEW

File: torch/distributed/pipelining/_unflatten.py [hunk 1]
```diff
@@ -1,453 +1,8 @@
...
-
-            # The only remaining possibility is that we are in the right stack
-            # frame. Copy the node into this frame's graph and increment the node counter.
-            assert node_module_stack == self.module_stack
-            self.copy_node(node)
-            node_idx += 1
+from torch.export.unflatten import _ModuleFrame
 
 
 def _outline_submodules(orig_graph: torch.fx.Graph):
```
Comment (H-Huang) [inline torch/distributed/pipelining/_unflatten.py:8]: Any value of this just getting moved to torch.export.unflatten as well? Or is this pretty PP specific?

**Review by H-Huang — APPROVED** [review]
looks good!

File: torch/distributed/pipelining/_unflatten.py (hunk 1 above)
Comment (kwen2501) [reply torch/distributed/pipelining/_unflatten.py:8]: yeah, this is the one func that's special.

**Pull request closed without merging**